src/main/java/Duke.java

```diff
@@ -1,10 +1,116 @@
+            if (br.readLine()!= null) {
+                System.out.println("Retrieving list details...");
+                FileInputStream fileStream = new FileInputStream("duke.txt");
+                ObjectInputStream inStream = new ObjectInputStream(fileStream);
+
+                taskItems = (ArrayList<Task>)inStream.readObject();
+
+                fileStream.close();
+                inStream.close();
+            }
+        }catch(IOException e) {
+            System.out.println("IOException occurred");
+            e.printStackTrace();
+        }catch(ClassNotFoundException e) {
+            System.out.println("ClassNotFoundException occurred");
+            e.printStackTrace();
+        }
+
+        String logo = "____________________________________________________________\n"
+                + " Hello! I'm Duke\n"
+                + " What can I do for you?\n\n"
+                +"____________________________________________________________\n";
+        System.out.println(logo);
+        Scanner stringScanner = new Scanner(System.in);
+        String userQuery = stringScanner.nextLine();
+        String userQueryReturn = "";
+
+        while(!userQuery.equalsIgnoreCase("bye")){
+            if(userQuery.contains("done")) {
+                int doneIndex = Integer.parseInt(userQuery.substring(5));
+                taskItems.get(doneIndex - 1).setStatus(true);
+                userQueryReturn = "____________________________________________________________\n"
+                        + " Nice! I've marked this task as done: \n"
+                        + " [" + taskItems.get(doneIndex - 1).getTypeIcon() + "][" + taskItems.get(doneIndex - 1).getStatusIcon() + "] " + taskItems.get(doneIndex - 1).getDescription() + "\n"
+                        + "____________________________________________________________\n";
+                System.out.println(userQueryReturn);
+                saveToFile(taskItems);
+                userQuery = stringScanner.nextLine();
+            }else if(userQuery.contains("delete")){
+                int doneIndex = Integer.parseInt(userQuery.substring(7));
+                Task removedItem = taskItems.remove(doneIndex);
+                userQueryReturn = "____________________________________________________________\n"
+                        + " Noted. I've removed this task: \n"
+                        + "   [" + removedItem.getTypeIcon() + "][" + removedItem.getStatusIcon() + "] " + removedItem.getDescription() + "\n"
+                        + " Now you have "+ taskItems.size() + " tasks in the list.\n"
+                        + "____________________________________________________________\n";
+                System.out.println(userQueryReturn);
+                saveToFile(taskItems);
+                userQuery = stringScanner.nextLine();
+            }else if(userQuery.contains("todo")||userQuery.contains("deadline")||userQuery.contains("event")) {
```

Comment:
Be consistent in your use of whitespaces between operators though it's probably a good idea to include whitespace between operators (see line 82)